les/fetcher.go

```diff
@@ -42,14 +42,16 @@ const (
 type lightFetcher struct {
 	pm    *ProtocolManager
 	odr   *LesOdr
-	chain *light.LightChain
+	chain lightChain
 
-	lock            sync.Mutex // lock protects access to the fetcher's internal state variables except sent requests
-	maxConfirmedTd  *big.Int
-	peers           map[*peer]*fetcherPeerInfo
-	lastUpdateStats *updateStatsEntry
-	syncing         bool
-	syncDone        chan *peer
+	lock               sync.Mutex // lock protects access to the fetcher's internal state variables except sent requests
+	maxConfirmedTd     *big.Int
+	maxConfirmedTdLock sync.RWMutex // lock protects access to maxConfirmedTd
```

Comment:
Why do you need all these locks? Also, are you sure this is safe? The light fetcher is not a performance critical part of the code, lightFetcher.lock did protect everything, it was the safest and simplest locking scheme.

Reply:
It fixes dataraces between updateMaxConfirmedTd and nextRequest.